### PR TITLE
Feature/merge execute payload

### PR DESCRIPTION
**Pull request by garyschulte**
## PR description
* implements engine_executePayload
* rename and repurpose rayonism options & flag
* treat the merge as another consensus mechanism:
  * add a `merge` consensus gradle subproject 
  * create MergeBesuController
  * create MergeProtocolSchedule 
  * create MergeContext for merge-specific consensus context
  * create MergeHeaderValidationRuleFactory for merge specific header validation rules
  * create MergeUnfinalizedValidationRule, attempt to enforce finalized block rules for import

## Fixed Issue(s)
protocol-misc 476 
protocol-misc 485

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExecutePayload.java [hunk 1]
```diff
@@ -75,6 +80,14 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
     final ExecutionEngineNewBlockParameter blockParam =
         requestContext.getRequiredParameter(0, ExecutionEngineNewBlockParameter.class);
 
+    Object reqId = requestContext.getRequest().getId();
```
Comment (matkt) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExecutePayload.java:83]: Don't we know the object type here  ?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExecutePayload.java [hunk 2]
```diff
@@ -92,11 +105,10 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
     } catch (final RLPException | IllegalArgumentException e) {
       LOG.warn("failed to decode transactions from newBlock RPC");
       e.printStackTrace();
```
Comment (matkt) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExecutePayload.java:107]: maybe we need to remove this printStackTrace

File: consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import org.hyperledger.besu.ethereum.mainnet.ProtocolScheduleBuilder;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSpecAdapters;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSpecBuilder;
+import org.hyperledger.besu.ethereum.mainnet.feemarket.FeeMarket;
+
+import java.math.BigInteger;
+
+public class MergeProtocolSchedule {
+
+  private static final BigInteger DEFAULT_CHAIN_ID = BigInteger.valueOf(1);
```
Comment (matkt) [inline consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java:31]: Why not using the network ID from this class `EthNetworkConfig`

File: consensus/merge/build.gradle [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      'Implementation-Title': archiveBaseName,
+      'Implementation-Version': calculateVersion()
+      )
+  }
+}
+
+repositories { mavenCentral() }
+
+dependencies {
+  implementation project(':config')
```
Comment (matkt) [inline consensus/merge/build.gradle:33]: just a question that comes to mind. are we sure we need all these dependencies

File: besu/src/main/java/org/hyperledger/besu/cli/options/unstable/MergeOptions.java [hunk 5]
```diff
@@ -51,7 +49,8 @@ public void consumeParameters(
         final Stack<String> args,
         final CommandLine.Model.ArgSpec argSpec,
         final CommandLine.Model.CommandSpec commandSpec) {
-      RayonismOptions.setMergeEnabled(Boolean.parseBoolean(args.pop()));
+      org.hyperledger.besu.config.experimental.MergeOptions.setMergeEnabled(
```
Comment (matkt) [inline besu/src/main/java/org/hyperledger/besu/cli/options/unstable/MergeOptions.java:52]: I think we can use an import here

**Review by matkt — APPROVED** [review]
Some comments. I approve for when they will be fixed

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExecutePayload.java (hunk 1 above)
Comment (garyschulte) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExecutePayload.java:83]: We don't.  JsonRpcRequestId is a plain object for some reason 🤷

File: consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java (hunk 3 above)
Comment (garyschulte) [reply consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java:31]: Good idea.  All of the other ProtocolSchedules just use an explicit BigInteger.

File: consensus/merge/build.gradle (hunk 4 above)
Comment (garyschulte) [reply consensus/merge/build.gradle:33]: Trimmed the dependencies to minimal :+1:  We can add stuff back as it becomes necessary

File: besu/src/main/java/org/hyperledger/besu/cli/options/unstable/MergeOptions.java (hunk 5 above)
Comment (garyschulte) [reply besu/src/main/java/org/hyperledger/besu/cli/options/unstable/MergeOptions.java:52]: Oh man yeah.  That is Intellij refactoring gone wild :)

**Pull request merged**